python/cudf/cudf/tests/test_joining.py

```diff
@@ -51,24 +51,43 @@ def make_params():
     aa = np.random.random(50)
     bb = np.random.random(50)
     for how in hows:
-        if how in ["left", "inner", "right"]:
+        if how in ["left", "inner", "right", "leftanti", "leftsemi"]:
             for method in methods:
                 yield (aa, bb, how, method)
         else:
             yield (aa, bb, how, "sort")
 
 
+def pd_odd_joins(left, right, join_type):
+    if join_type == "leftanti":
+        result = left.merge(
+            right,
+            how="left",
+            left_index=True,
+            right_index=True,
+            indicator=True,
+        )
+        return result.loc[result._merge == "left_only", :][left.columns]
+    elif join_type == "leftsemi":
+        return left[left.index.isin(right.index)][left.columns]
```

Comment:
I think anti can just be `left[~left.index.isin(right.index)][left.columns]` to simplify things as semi joins and anti joins are complements.